src/Entity/Post.php

```diff
@@ -85,4 +90,34 @@ public function setCreatedAt(\DateTimeImmutable $created_at): self
 
         return $this;
     }
+
+    /**
+     * @return Collection<int, Comment>
+     */
+    public function getComments(): Collection
+    {
+        return $this->comments;
+    }
+
+    public function addComment(Comment $comment): self
+    {
+        if (!$this->comments->contains($comment)) {
+            $this->comments[] = $comment;
+            $comment->setPost($this);
+        }
```

Comment:
Have a look at how I do this for packs. The ArrayCollection type is quite powerful and allows for a more declarative way to express this